### PR TITLE
feat: s3 prefix model data for JumpStartModel

### PR DESCRIPTION
**Pull request by evakravi**
*Description of changes:*
In order to allow for uncompressed model artifacts to directly be inputted into the `JumpStartModel` class, support was added to convert s3 prefix inputs into a `S3DataSource` dictionary. 

*Testing done:*
Unit tests added.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/factory/model.py [hunk 1]
```diff
@@ -218,6 +216,23 @@ def _add_model_data_to_kwargs(kwargs: JumpStartModelInitKwargs) -> JumpStartMode
         sagemaker_session=kwargs.sagemaker_session,
     )
 
+    if isinstance(model_data, str) and model_data.startswith("s3://") and model_data.endswith("/"):
+        if kwargs.model_data:
+            JUMPSTART_LOGGER.info(
+                "S3 prefix model_data detected for JumpStartModel: '%s'. "
+                "Converting to S3DataSource dictionary.",
```
Comment (whittech1) [inline src/sagemaker/jumpstart/factory/model.py:223]: nit: we could output the S3DataSource dictionary here as well for full visibility that customers can map back to the API docs for Hosting

File: tests/unit/sagemaker/jumpstart/constants.py [hunk 2]
```diff
@@ -1047,6 +1047,93 @@
...
+        "min_sdk_version": "2.130.0",
+        "training_supported": False,
+        "incremental_training_supported": False,
+        "hosting_ecr_specs": {
+            "framework": "pytorch",
+            "framework_version": "1.12.0",
+            "py_version": "py38",
+            "huggingface_transformers_version": "4.17.0",
+        },
+        "hosting_artifact_key": "huggingface-infer/infer-huggingface-text2text-flan-t5-xxl-fp16.tar.gz",
```
Comment (whittech1) [inline tests/unit/sagemaker/jumpstart/constants.py:1063]: nit: should probably change to a prefix here as well for an accurate test

**Review by whittech1 — APPROVED** [review]

**Review by whittech1 — APPROVED** [review]

**Review by JGuinegagne — APPROVED** [review]

**Review by whittech1 — APPROVED** [review]

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run pr

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by vpbhargav — APPROVED** [review]

**Review by benieric — COMMENTED** [review]
/bot run pr

**Comment by benieric** [comment]
PR Check: FAILED tests/integ/sagemaker/jumpstart/retrieve_uri/test_transfer_learning.py::test_jumpstart_transfer_learning_retrieve_functions

Is this related to any of these changes? Same test failed twice in a row.

**Review by benieric — COMMENTED** [review]
/bot run pr

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run pr

**Comment by mufaddal-rohawala** [comment]
The failing integ test is a flaky failure due to training job name conflicts. A fix is expected in subsequent PR from partner team.

**Pull request merged**